routes/auth.routes.js

```diff
@@ -0,0 +1,71 @@
+const { Router } = require("express");
+const router = new Router();
+const mongoose = require("mongoose");
+const User = require("../models/User.model");
+const bcryptjs = require("bcryptjs");
+const saltRounds = 10;
+const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard");
+
+router.get("/signup", isLoggedOut, (req, res) => res.render("auth/signup"));
+
+router.post("/signup", (req, res, next) => {
+  const { username, password } = req.body;
+
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({
+        username,
+        password: hashedPassword,
+      });
+    })
+    .then((userFromDB) => {
+      console.log("Newly created user is: ", userFromDB);
+    })
+    .catch((error) => next(error));
+});
+
+router.get("/login", isLoggedOut, (req, res) => res.render("auth/login"));
+
+router.post("/login", (req, res, next) => {
+  const { username, password } = req.body;
+
+  if (username === "" || password === "") {
+    res.render("auth/login", {
+      errorMessage: "Please enter both, username and password to login.",
+    });
+    return;
+  }
+
+  User.findOne({ username })
+    .then((user) => {
+      if (!user) {
+        res.render("auth/login", { errorMessage: "User not found" });
+        return;
+      } else if (bcryptjs.compareSync(password, user.password)) {
+        req.session.currentUser = user;
+        req.app.locals.currentUser = user;
+        res.render("profile", { user });
+      } else {
+        res.render("auth/login", { errorMessage: "Incorrect password" });
+      }
+    })
+    .catch((err) => next(err));
+});
+
+router.get("/private", isLoggedIn, (req, res) => {
+  res.render("user/private", { userInSession: req.session.currentUser });
+});
+
+router.get("/", (req, res) => res.render("user/main"));
```

Comment:
`router.get("/")` is the same as your homepage / index route, the user will never be able to see this page because it will always fall on the `router.get("/")` that is located on index.js

It should be something like
`router.get("/main", (req, res) => res.render("main"));`